sbi/inference/trainers/nle/nle_a.py

```diff
@@ -21,7 +22,7 @@ class NLE_A(LikelihoodEstimatorTrainer):
     def __init__(
         self,
         prior: Optional[Distribution] = None,
-        density_estimator: Union[str, Callable] = "maf",
+        density_estimator: Union[str, DensityEstimatorBuilder] = "maf",
```

Comment:
Please update the docstring here and in all other locations saying that the function provided should follow the `DensityEstimatorBuilder` protocol.